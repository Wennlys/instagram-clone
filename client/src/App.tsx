import React, { useEffect, useState } from 'react';
import Homepage from './pages/Homepage';
import Login from './pages/Login';
import api from './services/api';

function App(): JSX.Element {
  const [isLogged, setIsLogged] = useState(false);
  useEffect(() => {
    async function handleIsLogged() {
      const { data } = await api.post('/sessions', {
        user_id: 1,
        password: 12345678,
      });
      console.log(data);
      if (data[0].statusCode == 200) {
        return true;
      }
      return false;
    }

    handleIsLogged().then(res => setIsLogged(res));
  }, [isLogged]);

  return <div className="App">{isLogged ? <Homepage /> : <Login />}</div>;
}

export default App;
