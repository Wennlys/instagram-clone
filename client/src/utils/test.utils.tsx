import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

const renderWithRouter = (component: JSX.Element, initialRoute = '/'): RenderResult => {
  return render(<MemoryRouter initialEntries={[initialRoute]}>{component}</MemoryRouter>);
};

/* ---------- MOCKS ----------- */
const sessionsResponseMock = {
  post: {
    success: {
      status: 200,
      data: {
        token: 'asdfasdfasdfasdfasdfasdfasdf',
      },
    },
    failure: {
      status: 404,
      data: {
        error: {
          type: 'RESOURCE_NOT_FOUND',
          description: 'Wrong password, try again.',
        },
      },
    },
  },
};

const postsResponseMock = {
  get: {
    success: {
      status: 200,
      data: [
        {
          id: 1,
          image: 'http://localhost:3333/tmp/avatar.jpg',
          description: 'Nothing to see here. :P',
          userName: 'user1',
        },
        {
          id: 2,
          image: 'http://localhost:3333/tmp/avatar.jpg',
          description: 'Nothing to see here. :P',
          userName: 'user2',
        },
      ],
    },
  },
};

export { renderWithRouter, sessionsResponseMock, postsResponseMock };
