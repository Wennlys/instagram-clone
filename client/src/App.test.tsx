import '@testing-library/jest-dom/extend-expect';
import { cleanup, screen } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';
import App from './App';
import api from './services/api';
import { renderWithRouter } from './utils/test.utils';

jest.mock('./services/api');
const mockedApi = api as jest.Mocked<typeof api>;

afterEach(() => {
  cleanup();
  jest.resetAllMocks();
});

describe('App rendering & routes', () => {
  it('renders Login component', async () => {
    mockedApi.post.mockImplementation(() =>
      Promise.resolve({
        data: [
          {
            statusCode: 404,
          },
        ],
      }),
    );
    const { getByText } = renderWithRouter(<App />);
    expect(getByText(/Log In/i)).toBeInTheDocument();
  });

  test("authentication & Homepage's component rendering", async () => {
    mockedApi.post.mockImplementation(() =>
      Promise.resolve({
        data: [
          {
            statusCode: 200,
          },
        ],
      }),
    );
    await act(async () => {
      await renderWithRouter(<App />);
      (await screen.findByText(/Log In/i)).click();
    });
    expect(screen.getByText(/Home/i)).toBeInTheDocument();
  });

  it('renders 404 component', async () => {
    mockedApi.post.mockImplementation(() =>
      Promise.resolve({
        data: [
          {
            statusCode: 404,
          },
        ],
      }),
    );
    renderWithRouter(<App />, '/404');
    expect(screen.getByText(/Not Found/i)).toBeInTheDocument();
  });

  it('renders Explore component', async () => {
    mockedApi.post.mockImplementation(() =>
      Promise.resolve({
        data: [
          {
            statusCode: 200,
          },
        ],
      }),
    );
    await act(async () => {
      await renderWithRouter(<App />);
      (await screen.findByText(/Log In/i)).click();
      (await screen.findByText(/Explore/i)).click();
    });
    expect(screen.getByText(/Explore/i)).toBeInTheDocument();
  });
});
