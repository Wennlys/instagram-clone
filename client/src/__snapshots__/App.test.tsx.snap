// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Login"
    >
      login
    </div>
  </div>
</DocumentFragment>
`;
